src/room.js

```diff
@@ -0,0 +1,37 @@
+class Room {
+
+  constructor (roomname) {
+    this.roomname = roomname;
+    this.endpoints = {};
+  }
+  addEndpoint (endpointId) {
+    this.endpoints[endpointId] = {
+      username: '',
+      permissions: 'CHAT',
+      commsId: ''
+    };
+  }
+  updateUsernameame (endpointId, username) {
+    this.endpoints[endpointId].username = username;
+  }
+  updatePermissions (endpointId, permissions) {
+    this.endpoints[endpointId].permissions = permissions;
+  }
+  updateCommsId (endpointId, commsId) {
+    this.endpoints[endpointId].commsId = commsId;
+  }
+  removeEndpoint (endpointId) {
+    delete this.endpoints[endpointId];
+  }
+  getRoomName () {
+    return this.roomname;
+  }
+  getEndpointNameFromCommsID (commsId) {
+    for (let props in this.endpoints) {
+      if (this.endpoints[props].commsId === commsId) {
+        return props;
+      }
+    }
```

Comment:
We should also have a getCommsId method, that takes an endpointId as its argument and returns the endpoints comms id